File: RentABikeWTR_v1_7.Model/ServisiranjaPregled.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RentABikeWTR_v1_7.Model
{
    public class ServisiranjaPregled
    {
        public int ServisiranjeId { get; set; }
        public string OpisKvara { get; set; }
        public DateTime DatumServisiranja { get; set; }

        public string? PreduzetaAkcija { get; set; }
        public string? KomentarServisera { get; set; }
        public int? BiciklID { get; set; }
        public string? NazivBicikla { get; set; }
        public List<RezervniDijelovi>? RezervniDijelovi { get; set; }
    }
}

File: rentabikewtr_desktop/lib/model/servisiranjaPregled.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';
import 'package:rentabikewtr_desktop/model/rezervniDijeloviPregled.dart';

part 'servisiranjaPregled.g.dart';

@JsonSerializable()
class ServisiranjaPregled {
  int? servisiranjeId;
  String? opisKvara;
  DateTime? datumServisiranja;
  String? preduzetaAkcija;
  String? komentarServisera;
  String? nazivBicikla;
  int? biciklID;
  List<RezervniDijeloviPregled>? rezervniDijelovi;

  ServisiranjaPregled() {}

  factory ServisiranjaPregled.fromJson(Map<String, dynamic> json) =>
      _$ServisiranjaPregledFromJson(json);

  /// Connect the generated [_$ServisiranjaPregledToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ServisiranjaPregledToJson(this);
}
